patches/runpatches.sh

```diff
@@ -31,21 +30,67 @@ curl https://android.googlesource.com/platform/art/+/master/openjdkjvmti/include
 
 # patch tools/base
 cd "${SRC}/tools/base"
+git reset --hard
 git apply "${PATCHES}/tools.base.bazel.patch"
 
 # patch tools/adt/idea
 cd "${SRC}/tools/adt/idea"
+git reset --hard
 git apply "${PATCHES}/tools.adt.idea.patch"
 
+# patch external/grpc for Mac laptops
+cd "${SRC}/external/grpc-grpc"
+git reset --hard
+git apply "${PATCHES}/external.grpc-grpc.patch"
+
 ## setup intellij-sdk (this can be replaced with IC-211 + a few libraries updated)
-rm -rf "${SRC}/prebuilts/studio/intellij-sdk/AI-221/linux" || true
-mkdir -p "${SRC}/prebuilts/studio/intellij-sdk/AI-221/linux"
-cd "${SRC}/prebuilts/studio/intellij-sdk"
-echo "d1b657c78629333b4b027c2e05a49a353dc624d7d89ea0836a7c11f90eeac947 android-studio-2022.1.1.8-linux.tar.gz" > android-studio-2022.1.1.8-linux.tar.gz.sha256
-if ! sha256sum -c android-studio-2022.1.1.8-linux.tar.gz.sha256; then
-    curl -L https://r1---sn-5hneknes.gvt1.com/edgedl/android/studio/ide-zips/2022.1.1.8/android-studio-2022.1.1.8-linux.tar.gz > android-studio-2022.1.1.8-linux.tar.gz
-    sha256sum -c android-studio-2022.1.1.8-linux.tar.gz.sha256
+UNAME_OUT=$(uname)
+if [ $UNAME_OUT == "Darwin" ]; then
+    OS="darwin"
+else
+    OS="linux"
+fi
+
+IDEA_VER=AI-222
```

Comment:
oh-ah. This script has been significantly reworked in our internal sources. But idea is clear - use Darwin on Mac